### DISCUSSION TITLE
[Node Operator Question] get latest TXs (fastest way(Mempool)) by running op-node

### DISCUSSION THREAD

**Discussion by ZigBalthazar** [ethereum-optimism/developers#327] [Node Operators] [answered]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi

I run an op-node with this command :

```
.\bin\op-node.exe --l1=<alchemy> --l1.rpckind=alchemy --l1.trustrpc=true --l2=<alchemy>  --network=op-mainnet --syncmode=execution-layer --l2.jwt-secret=./jwt --l1.beacon.ignore=true
```

my goal is to retrieve the latest Txs on L2 in fatest way(from the Sequencer before those blocks are submitted to L1)

after the op-node ran I successful called this Curl and get a response from my op-node :

```
curl -X POST -H "Content-Type: application/json" --data \
    '{"jsonrpc":"2.0","method":"optimism_version","params":[],"id":1}' \
    http://localhost:9545
```

I need to use this rpc :
https://docs.optimism.io/builders/app-developers/transactions/statuses#sequencer-confirmed--unsafe

what should i do now?

note : 
I did not run op-geth, I use alchemy for L1 and L2
`8545` is the geth port
I want to use op-node rpc(default port : `9545`)

### Protocol Description

.

**Comment by Chomtana** [comment, marked as answer]
Optimism is featuring a private mempool. The fastest way is probably listening through P2P gossip that I have guided in the previous topic. But is probably hard to do.

https://github.com/ethereum-optimism/developers/discussions/312#discussioncomment-8964397

I have just come up with another method to modify the op-node gossip handler function to send the tx to your application which may be easier. Then you could run that modified op-node and transaction will be streamed to your application.

**Reply by ZigBalthazar** [reply]
which function is responsible for handling the gossip income messages?
